### ISSUE TITLE
Update on recent Faust version

### ISSUE THREAD

**Issue by krejcmat** [marcosschroh/python-schema-registry-client#44]
I have tried the python-schema-registry-client with latest Faust v1.10.3 and seems it is not compatible. 
Last compatible version of Faust seems to be v1.10.0. From that version, Faust is crashing on startup 
```
 [^-App]: Crashed reason=TypeError("argument of type 'Stream' is not iterable") 
Traceback (most recent call last):
  File "fastavro/_write.pyx", line 267, in fastavro._write.write_record
TypeError: Expected dict, got Stream
```

**Comment by marcosschroh** [comment]
Hi @krejcmat 

Thanks for reporting the issue. The requirement is `faust<=1.9.0`. I will try to update the library in order to use the latest `faust` version. Can you share the code that you were trying to execute?

**Comment by krejcmat** [comment]
Hi @marcosschroh,
it is related to Model optimization [described in ](https://github.com/robinhood/faust/compare/v1.9.0...v1.10.0#diff-74df3e650912486104daef2fd7518b53R75)

specifically, the error I have posted above is related to [this](https://github.com/robinhood/faust/compare/v1.9.0...v1.10.0#diff-74df3e650912486104daef2fd7518b53R116). 
It is a bit naming confusion because I named "Stream" one of my Record class.
```
class Registry(faust.Record, serializer='registry'):
    input: Stream

class Stream(faust.Record, serializer='stream',):
    schema: str
    topic: str
```

**Comment by krejcmat** [comment]
> I have tried the python-schema-registry-client with latest Faust v1.10.3 and seems it is not compatible.
> Last compatible version of Faust seems to be v1.10.0. From that version, Faust is crashing on startup
> 
> ```
>  [^-App]: Crashed reason=TypeError("argument of type 'Stream' is not iterable") 
> Traceback (most recent call last):
>   File "fastavro/_write.pyx", line 267, in fastavro._write.write_record
> TypeError: Expected dict, got Stream
> ```

Last compatible version is 1.9.0

**Comment by marcosschroh** [comment]
Hi @krejcmat ,

I have checked with the latest faust version (`1.10.4`) and the library seems to work properly. Here I have a project [example](https://github.com/marcosschroh/faust-docker-compose-example).

Please take a look the [UserModel](https://github.com/marcosschroh/faust-docker-compose-example/blob/master/faust-project/example/users/models.py) which is using `avro_user_serializer` (`FaustSerializer` from `python-schema-registry-client`)

Let me know if you have any questions

**Comment by krejcmat** [comment]
> Hi @krejcmat ,
> 
> I have checked with the latest faust version (`1.10.4`) and the library seems to work properly. Here I have a project [example](https://github.com/marcosschroh/faust-docker-compose-example).
> 
> Please take a look the [UserModel](https://github.com/marcosschroh/faust-docker-compose-example/blob/master/faust-project/example/users/models.py) which is using `avro_user_serializer` (`FaustSerializer` from `python-schema-registry-client`)
> 
> Let me know if you have any questions

Yes, the library works if is not used the scenario I posted above. Where the Model has a variable of the type of another Model. I had this case in my codebase where I am using your plugin and until  v1.9.0 it worked. I cannot tell if this issue has to be solved in this plugin or in Faust. 
I work around it by overloading to_representation method.
```
class Registry(faust.Record, serializer='registry'):
    input: Stream
    output: Stream

    def to_representation(self):
        payload = self.asdict()
        payload['input'] = payload['input'].to_representation()
        payload['output'] = payload['output'].to_representation()
        options = self._options
        if options.include_metadata:
            payload['__faust'] = {'ns': options.namespace}
        return payload
```

**Comment by marcosschroh** [comment]
Make sense what you are doing because `faust` changed the `_BUILD_asdict_field`. Here is the diff: [version 1.9.0](https://github.com/robinhood/faust/blame/v1.9.0/faust/models/record.py#L594) and [current version](https://github.com/robinhood/faust/blame/38d61a78c76a096597421a8e9db2878d4381dd6a/faust/models/record.py#L564).

I have created a [PR](https://github.com/marcosschroh/python-schema-registry-client/pull/45/files) to update the `Faust Serializer` in order to use the latest Faust version

**Comment by marcosschroh** [comment]
fixed by https://github.com/marcosschroh/python-schema-registry-client/pull/45

**Comment by krejcmat** [comment]
> fixed by #45

That is great. Thanks
